plugins/wasmedge_httpsreq/httpsreqfunc.cpp

```diff
@@ -15,62 +15,57 @@
 #include <unistd.h>
 
-// Some of the code was taken from this post:
-// https://stackoverflow.com/questions/52727565/client-in-c-use-gethostbyname-or-getaddrinfo
-
 namespace WasmEdge {
 namespace Host {
 
-Expect<void> SendData::body(const Runtime::CallingFrame &Frame,
-                            uint32_t HostPtr, uint32_t HostLen, uint32_t Port,
-                            uint32_t BodyPtr, uint32_t BodyLen) {
+Expect<void> WasmEdgeHttpsReqSendData::body(const Runtime::CallingFrame &Frame,
+                                            uint32_t HostPtr, uint32_t HostLen,
+                                            uint32_t Port, uint32_t BodyPtr,
+                                            uint32_t BodyLen) {
   auto *MemInst = Frame.getMemoryByIndex(0);
   if (MemInst == nullptr) {
     return Unexpect(ErrCode::Value::HostFuncError);
   }
 
-  char *Host = MemInst->getPointer<char *>(HostPtr);
-  std::string NewHost;
-  std::copy_n(Host, HostLen, std::back_inserter(NewHost));
-  Env.Host = std::move(NewHost);
-
-  Env.Port = Port;
-
-  char *BodyStr = MemInst->getPointer<char *>(BodyPtr);
-  std::string NewBodyStr;
-  std::copy_n(BodyStr, BodyLen, std::back_inserter(NewBodyStr));
-  Env.BodyStr = std::move(NewBodyStr);
+  const char *Host = MemInst->getPointer<const char *>(HostPtr);
+  const char *Body = MemInst->getPointer<const char *>(BodyPtr);
+  if (Host == nullptr) {
+    spdlog::error("[Wasmedge Httpsreq] Fail to get Host");
+    return Unexpect(ErrCode::Value::HostFuncError);
+  }
+  if (Body == nullptr) {
+    spdlog::error("[Wasmedge Httpsreq] Fail to get Body");
+    return Unexpect(ErrCode::Value::HostFuncError);
+  }
+  std::string HostStr, BodyStr, PortStr = std::to_string(Port);
+  std::copy_n(Host, HostLen, std::back_inserter(HostStr));
+  std::copy_n(Body, BodyLen, std::back_inserter(BodyStr));
 
   const SSL_METHOD *Method = TLS_client_method();
   SSL_CTX *Ctx = SSL_CTX_new(Method);
-
   if (Ctx == nullptr) {
     ERR_print_errors_fp(stderr);
-    exit(EXIT_FAILURE);
+    spdlog::error("[Wasmedge Httpsreq] SSL_CTX_new() failed");
+    return Unexpect(ErrCode::Value::HostFuncError);
   }
-
   SSL *Ssl = SSL_new(Ctx);
   if (Ssl == nullptr) {
-    fprintf(stderr, "[Httpsreq plugin] SSL_new() failed\n");
-    exit(EXIT_FAILURE);
+    spdlog::error("[Wasmedge Httpsreq] SSL_new() failed");
+    return Unexpect(ErrCode::Value::HostFuncError);
   }
 
   // open connection
   int Sfd, Err;
   struct addrinfo Hints = {}, *Addrs;
-  char PortStr[16] = {};
 
   Hints.ai_family = AF_INET;
   Hints.ai_socktype = SOCK_STREAM;
   Hints.ai_protocol = IPPROTO_TCP;
 
-  std::sprintf(PortStr, "%d", Port);
-
-  Err = getaddrinfo(Env.Host.c_str(), PortStr, &Hints, &Addrs);
+  Err = getaddrinfo(HostStr.c_str(), PortStr.c_str(), &Hints, &Addrs);
   if (Err != 0) {
-    fprintf(stderr, "[Httpsreq plugin] %s: %s\n", Env.Host.c_str(),
-            gai_strerror(Err));
-    abort();
+    spdlog::error("[WasmEdge Httpsreq] {}", gai_strerror(Err));
```

Comment:
You remind me that we should use `WasmEdge` not `Wasmedge` in other places. Please check it again.